packages/dynamite/dynamite_end_to_end_test/lib/type_defs.openapi.dart

```diff
@@ -58,30 +58,31 @@ abstract class NestedRedirect implements $NestedRedirectInterface, Built<NestedR
 }
 
 typedef SomeOfRedirect = ({Base? base, int? $int, JsonObject? jsonObject});
-typedef $BaseIntJsonObject = ({Base? base, int? $int, JsonObject? jsonObject});
+typedef $e1c7ecea8e5fdae7b94cd86c0dc4f1ba = ({Base? base, int? $int, JsonObject? jsonObject});
```

Comment:
As one can see here there is always going to be a typedef with a usable name.
The hashed name is only needed for the serializer to match all typedefs with those usable names.

Reply:
I disagree about the "usable". Also in the generate docs, the names would be meaningless. There's no way we can push this onto the package consumers.

Reply:
I'd say not having a typedef is more user friendly in this case.

Reply:
The problem is that we need the typedef for the serializer as it can not work with raw records (the runtimetype of them is not constant while the one of a typedef is. So it also has performance implications).

Therefore we generate a typedef for use by the devs (`SomeOfRedirect` and one to use by the serializer `$e1c7ecea8e5fdae7b94cd86c0dc4f1ba`).
A dev should never need to use the later one as everything (other than the fromJson method that's static to the extension).

Reply:
If the fromJson reviver is that important We can generate extensions for the human readable version to only handle that while having a common extension (on the hashed name) that handles the non static methods.

Reply:
And if you are curious we can not generate extension methods with the non static methods for every typedef as ones this would lead to conflicts between someOfs with the same type.

Reply:
Ok, but how would you generate human readable extension names that are not conflicting and not too long?

Reply:
We already do generate human readable ones.
That's the whole point.

The not human readable ones are not needed by the devs. I just made them private and will push these changes later. Maybe this helps in understanding.

Reply:
Sorry, I didn't find the time earlier.
Does it make sense now?